core/src/main/scala/quasar/datasource/kafka/Config.scala

```diff
@@ -28,14 +28,22 @@ case class Config(
   bootstrapServers: NonEmptyList[String],
   groupId: String,
   topics: NonEmptyList[String],
+  tunnelConfig: Option[TunnelConfig],
   decoder: Decoder,
   format: DataFormat
 ) extends Product with Serializable {
   def isTopic(topic: String): Boolean = topics.exists(_ == topic)
 
-  // No sensitive information at this time
-  def sanitize: Config = this
-  def reconfigure(patch: Config): Either[Config, Config] = patch.asRight
+  def sanitize: Config = copy(tunnelConfig = tunnelConfig.map(_.sanitize))
+  def reconfigure(patch: Config): Either[Config, Config] = {
+    val reconfigured = (tunnelConfig, patch.tunnelConfig) match {
+      case (_, Some(newTc)) if newTc.auth.nonEmpty => Left(newTc.sanitize)
+      case (Some(tc), Some(newTc)) => Right(Some(newTc.copy(auth = tc.auth)))
+      case (None, other) => Right(other)
+      case (_, None) => Right(None)
```

Comment:
The two cases
```scala
      case (None, other) => Right(other)
      case (_, None) => Right(None)
```
can be condensed into a single case
```scala
      case (_, other) => Right(other)
```

Reply:
Done